provers/mock/prover.go

```diff
@@ -117,10 +118,31 @@ func (pr *Prover) CheckRefreshRequired(dst core.ChainInfoICS02Querier) (bool, er
 
 // ProveState returns the proof of an IBC state specified by `path` and `value`
 func (pr *Prover) ProveState(ctx core.QueryContext, path string, value []byte) ([]byte, clienttypes.Height, error) {
-	return makeProof(value), ctx.Height().(clienttypes.Height), nil
+	height := ctx.Height().(clienttypes.Height)
+	return makeProof(height, path, value), height, nil
 }
 
-func makeProof(bz []byte) []byte {
-	h := sha256.Sum256(bz)
+func makeProof(height exported.Height, path string, bz []byte) []byte {
+	revisionNumber := height.GetRevisionNumber()
+	revisionHeight := height.GetRevisionHeight()
+
+	heightBig := new(big.Int)
+	revisionNumberBig := new(big.Int).SetUint64(revisionNumber)
+	revisionNumberBig = revisionNumberBig.Lsh(revisionNumberBig, 64)
+	revisionHeightBig := new(big.Int).SetUint64(revisionHeight)
+	heightBig.Or(revisionNumberBig, revisionHeightBig)
+	hashHeight := sha256.Sum256(heightBig.Bytes())
```

Comment:
The new spec is `sha256(height ++ sha256(prefix) ++ sha256(path) ++ sha256(value))`.
You should not hash the height here.

I think you can use `encoding/binary` to do this easilier.
https://pkg.go.dev/encoding/binary#example-ByteOrder-Put

Reply:
Use encoding/binary
https://github.com/hyperledger-labs/yui-relayer/pull/129/commits/309fad3580e668febcf3c3b9a5f40124e7bc69c4